packages/ui/core-components/src/lib/unsorted/viz/map/EvidenceMap.js

```diff
@@ -510,7 +526,26 @@ export class EvidenceMap {
 			values = this.handleLegendValues(colorPalette, values, legendType);
 			this.buildLegend(colorPalette, values, minValue, maxValue);
 		}
+
 		// Return the values, minValue, and maxValue for sharing with other functions
-		return { values, minValue, maxValue, colorScale, colorPalette };
+		return { values, minValue, maxValue, colorScale, colorPalette, paneType };
+	}
+
+	/**@type {[string]} */
+	#paneArray = [];
+
+	checkPanes(paneType) {
+		let newPaneType = paneType;
+		let suffix = 1;
+
+		// Continue appending "-suffix" until a unique paneType is found
+		while (this.#paneArray.includes(newPaneType)) {
+			newPaneType = `${paneType}-${suffix}`;
+			suffix += 1;
+		}
+
+		// Add the unique paneType to the array
+		this.#paneArray.push(newPaneType);
+		return newPaneType;
 	}
```

Comment:
Is `paneType` used for anything? Can we get by with just the `suffix` - an integer that increments for each pane? Or maybe a `nanoid`?

Reply:
Yeah nanoid should work,

I did enjoy using the paneType to easily identify the DOM rendering pattern

![image](https://github.com/user-attachments/assets/c45e1012-b68c-4302-a106-4a63e24e69d3)

If i were to come back to it, it would be easier for me to know which pane layer was which with the naming convention

Reply:
Generating IDs in a while loop is generally not a great idea. If for some reason the logic broke (e.g. by a future change) it would cause an infinite loop and would lock everything.

We can keep pane type for debugging, but I'd prefer to use `nanoid` to ensure uniqueness instead of the loop.

Reply:
Sounds good, will change to nanoid!